### ISSUE TITLE
Improve landmark labeling

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2472]
In instances of multiple landmarks of the same type, landmarks should have proper labeling to improve page navigation / semantics. Existing labels could also be made more specific in some cases.

Admin side, this is a common occurrence with coexisting navigation landmarks. The navigation menu is labeled as "Navigation menu", but nested site-specific navigation menus are not labeled. Page section tabs are contained within navigation landmarks, but those are also unlabeled.

**Comment by kimisgold** [comment]
I've begun improving the labeling for our admin landmarks on the branch `landmark-labels`. This is the list of landmarks I've found thus far, along with the values I've provided for their `aria-label`s.

* [x] `<header>` - "Header"
* [x] `<main>` - "Main content"
* [x] `<footer>` - "Footer"
* [x] `role="search"` - "Search items", "Search item sets", "Search media"
* [ ] `<nav>`
  * [x] Admin sidebar
    * [x] Installation admin - "[Installation title] admin"
    * [x] Site-level admin - "[Site title] admin"
  * [ ] Site pages
    * [ ] Breadcrumbs - "Breadcrumbs"
    * [ ] Child pages - "Child pages"
  * [x] Form sidebar pagination
    * [x] Search query preview - "Search results pagination"
    * [x] Item - "Item search results pagination"
    * [x] Item set - "Item set search results pagination"
    * [x] Media - "Media search results pagination"
    * [x] Asset - "Asset search results pagination"

Let me know if there's anything I missed. These labels are also open for further refinement, happy to take suggestions.

**Comment by aviannamiller** [comment]
I cannot access site pages; the following error displays:
```
Error: Call to undefined function __() in /var/www/html/amiller/OmekaS/application/view/omeka/site/page/show.phtml:22
Stack trace:
#0 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-view/src/Renderer/PhpRenderer.php(585): include()
#1 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-view/src/View.php(202): Laminas\View\Renderer\PhpRenderer->render(NULL)
#2 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-view/src/View.php(230): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
#3 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-view/src/View.php(195): Laminas\View\View->renderChildren(Object(Laminas\View\Model\ViewModel))
#4 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-mvc/src/View/Http/DefaultRenderingStrategy.php(92): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
#5 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-eventmanager/src/EventManager.php(318): Laminas\Mvc\View\Http\DefaultRenderingStrategy->render(Object(Laminas\Mvc\MvcEvent))
#6 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-eventmanager/src/EventManager.php(168): Laminas\EventManager\EventManager->triggerListeners(Object(Laminas\Mvc\MvcEvent))
#7 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-mvc/src/Application.php(354): Laminas\EventManager\EventManager->triggerEvent(Object(Laminas\Mvc\MvcEvent))
#8 /var/www/html/amiller/OmekaS/vendor/laminas/laminas-mvc/src/Application.php(335): Laminas\Mvc\Application->completeRequest(Object(Laminas\Mvc\MvcEvent))
#9 /var/www/html/amiller/OmekaS/index.php(26): Laminas\Mvc\Application->run()
#10 {main}
```

**Comment by aviannamiller** [comment]
Similar issue with sidebar pagination:
```
Warning: Undefined variable $paginationType in /var/www/html/amiller/OmekaS/application/view/common/sidebar-pagination.phtml on line 3
```

**Comment by aviannamiller** [comment]
Thanks for the work on this! Things are looking good so far. Before proceeding much further, I would say we should be careful not to add redundancy where it is not needed. From the list here, I would cut the labels on `<header>`, `<main>`, and `<footer>`. Similarly, words or phrases that reiterate the landmark name could be removed. One instance of this would be for the search. Currently, you get something like “Search items, search”. If we cut out "search" from each of these labels, the announcement would more succinctly just be “Items, search” since the landmark itself already contributes that context. What do you think?

I'm also noticing some styling issues with the login and site browse pages on this branch.
<img width="680" height="588" alt="Login page. Elements are styled against a dark background and have very low contrast." src="https://github.com/user-attachments/assets/a3efc286-a31f-490b-aa45-9bbcd5f19598" />

<img width="924" height="471" alt="Site browse page. Elements are styled against a dark background and have very low contrast." src="https://github.com/user-attachments/assets/8863ace2-320f-4233-b6ad-2b9c7d5080f8" />

I've looked around briefly, but I'm not positive about any other spots having similar unintended changes.

**Comment by kimisgold** [comment]
The pagination labeling should now be fixed. I had missed the minimal layout for updating some landmarks, hence the broken styles. They should now be consistent with the rest of the admin.

I've cut out most of the repetitive labels you've identified, with the exception of `<footer>`. The screenreader default readout seems to be "content". Maybe that's an expected pattern, but the readout of "Footer, content" doesn't seem to be as repetitive as the others.

**Comment by aviannamiller** [comment]
> I've cut out most of the repetitive labels you've identified, with the exception of `<footer>`. The screenreader default readout seems to be "content". Maybe that's an expected pattern, but the readout of "Footer, content" doesn't seem to be as repetitive as the others.

Assumedly coming from the `contentinfo` role, but yes, I think this is fine to accommodate the way different screen readers might handle these landmarks. I guess it's similar to the header/banner announcements. These are all areas (albeit smaller) that could always use more feedback down the line.

I'm blanking... can you please provide an example of where a "Search query preview - "'Search results pagination'" landmark might appear in a sidebar?

And the same issue persists for site pages with that error displaying.

Some other admin-side areas that are missing landmarks or labels:
- [ ] Admin front page - `<search>`
- [ ] Section navigation tabs, under `<nav>`
  - Perhaps the tab pattern needs some reworking anyway, but a label to indicate the presence of section tabs within the nav landmark would probably be suitable in the interim
- [ ] Sidebar search bars - `<search>`
- [ ] Sidebar filter [properties, item sets, etc.] inputs - (perhaps) `<search>`
- [ ] Potential landmarks for sidebars as a whole, as identified in https://github.com/omeka/omeka-s/issues/2339

**Comment by kimisgold** [comment]
> I'm blanking... can you please provide an example of where a "Search query preview - "'Search results pagination'" landmark might appear in a sidebar?

You can go to Site Admin > Resources > Add/Replace/Remove to activate the Search Query field. Once you build the query, the pagination will appear when you preview the search results.

Hmm, I would appreciate some suggestions for how to differentiate the search in the left admin bar, which is labeled "Items/Item Set/Media" depending on the resource selected, and a potential sidebar search. The sidebar search is often a resource search as well, looking for resources to assign in an interaction.

**Comment by aviannamiller** [comment]
> You can go to Site Admin > Resources > Add/Replace/Remove to activate the Search Query field. Once you build the query, the pagination will appear when you preview the search results.

Thank you!

In terms of the sidebar search, applying a landmark region to the sidebars may help to isolate any controls within the sidebars as a group, but in terms of a label for the searches, I've attempted to outline a few possibilities, which may have some degree of overlap between them:
- Provide a pretty simple label (so you end up with something like "Item set, search") to a newly added `<search>` landmark in the sidebar. Test out if providing a landmark role to the sidebar (aside/complementary? section/region?) itself provides enough context to distinguish the search as sidebar-specific.
- Assign the sidebar heading to the search label, so you get something like "Select item set, search". This gets a bit more complicated in cases like the sidebar for linking a new or existing item, in which case an equivalent label to pull from is within the navigation tabs, not the heading. Also, the wording of the label could imply that the search action itself makes the selection as opposed to it filtering down resource options.
- Aim for a simpler label (e.g. "Item set, search", as above), but also provide an `aria-description` to better indicate input purpose. For instance, "Search to filter resources available to be selected for internal linking" (there's definitely a more straightforward way to say this). I don't think `aria-description` is fully ready to go as of yet, so it's still recommended to use `aria-describedby`. If going this route, perhaps visible helper text that briefly indicates the expected input and can then be linked as the search description could be a good option.
